core/store/models/job_spec.go

```diff
@@ -43,15 +43,16 @@ type TaskSpecRequest struct {
 // for a given contract. It contains the Initiators, Tasks (which are the
 // individual steps to be carried out), StartAt, EndAt, and CreatedAt fields.
 type JobSpec struct {
-	ID         *ID          `json:"id,omitempty" gorm:"primary_key;not null"`
-	CreatedAt  time.Time    `json:"createdAt" gorm:"index"`
-	Initiators []Initiator  `json:"initiators"`
-	MinPayment *assets.Link `json:"minPayment,omitempty" gorm:"type:varchar(255)"`
-	Tasks      []TaskSpec   `json:"tasks"`
-	StartAt    null.Time    `json:"startAt" gorm:"index"`
-	EndAt      null.Time    `json:"endAt" gorm:"index"`
-	DeletedAt  null.Time    `json:"-" gorm:"index"`
-	UpdatedAt  time.Time    `json:"-"`
+	ID         *ID            `json:"id,omitempty" gorm:"primary_key;not null"`
+	CreatedAt  time.Time      `json:"createdAt" gorm:"index"`
+	Initiators []Initiator    `json:"initiators"`
+	MinPayment *assets.Link   `json:"minPayment,omitempty" gorm:"type:varchar(255)"`
+	Tasks      []TaskSpec     `json:"tasks"`
+	StartAt    null.Time      `json:"startAt" gorm:"index"`
+	EndAt      null.Time      `json:"endAt" gorm:"index"`
+	DeletedAt  null.Time      `json:"-" gorm:"index"`
+	UpdatedAt  time.Time      `json:"-"`
+	Errors     []JobSpecError `json:"errors" gorm:"foreignkey:JobSpecID"`
```

Comment:
Might want to disable autosave/autoupdate if it's not necessary. I find that to be a highly magical part of gorm that can cause surprising behaviour.